src/main/java/com/gregtechceu/gtceu/common/machine/electric/BatteryBufferMachine.java

```diff
@@ -52,24 +53,20 @@ public class BatteryBufferMachine extends TieredEnergyMachine
     @SaveField
     protected final CustomItemStackHandler batteryInventory;
 
-    public BatteryBufferMachine(IMachineBlockEntity holder, int tier, int inventorySize, Object... args) {
-        super(holder, tier, inventorySize);
+    public BatteryBufferMachine(BlockEntityCreationInfo info, int tier, int inventorySize) {
+        super(info, tier,
+                (TieredEnergyMachine machine) -> new EnergyBatteryTrait((BatteryBufferMachine) machine, inventorySize));
```

Comment:
Surely this can be made better in some way. Having to make a typed lambda parameter, and then also casting the value to another type in the body feels really jank.

Can this maybe be made into a `Supplier<T>`, locally capturing `this`? For example:
```java
() -> new EnergyBatteyTrait(this, inventorySize)
```